docs/debugging/ClickThePaintingsLv1-kr.md

```diff
@@ -0,0 +1,673 @@
+  return (
+    <NativeRouter> 
+      <ScrollView style={styles.container}>
+        <TopBar />
+        <Route path="/" component={ListView} /> 
+        <Route path="/detail/:id" component={DetailView} />
+      </ScrollView>
+    </NativeRouter>
+  );
+};
+
+...
+```
+
+먼저 라우터를 사용하기 위해 `NativeRouter`로 감쌉니다. 그리고 `Route` 컴포넌트를 사용하여 어떤 **path**로 접근하면 어떤 **component**를 보여줄지 설정합니다. 우리는 루트 경로(`/`)로 접근하면 ListView 컴포넌트를 보여주도록 하고, `/detail/:id`로 접근하면 해당 id의 DetailView를 보여주도록 설정합니다. 여기서 /detail 뒤에 붙는 /:id는 해당 URL에 id를 패러미터로 전달하는 문법입니다.
+
+예를 들어 
+1. Image 1을 클릭하면 `/detail/1` 이라는 주소로 라우팅하도록 합시다. 
+2. 그러면 /detail/ 뒤에 붙은 숫자 1이 패러미터로 전달될 것입니다. 이를 전달받아서 ID가 1인 아이템의 세부 정보를 보여주는 DetailView를 렌더링하면 됩니다.
+
+위에서 1번 작업을 하기 위해서 ListView에서 [Link](https://reacttraining.com/react-router/native/api/Link) 컴포넌트를 사용할 것입니다. a 태그와 동일한 기능을 하지만 리액트 라우터를 사용할 때에는 a 태그 대신 Link 컴포넌트를 사용해야 합니다. a 태그는 페이지를 새로 불러오면서 컴포넌트가 갖고 있던 상태를 모두 날려버리기 때문입니다. 
+
+2번 작업은 이미 [Route](https://reacttraining.com/react-router/native/api/Route) 컴포넌트로 DetailView를 만들었으니, 실제로 *DetailView.js*를 작성하여 id 패러미터가 잘 전달되는지 확인해 보겠습니다. 
+   
+이제 1번 작업부터 해보겠습니다. 
+
+## ListView에서 Link 컴포넌트 사용하기
+
+***ListView.js***
+
+```jsx
+...
+
+import {Link} from 'react-router-native';
+
+const deviceWidth = Dimensions.get('window').width;
+
+const ListView = () => {
+  const data = [1, 2, 3, 4, 5];
+
+  const imgList = data.map((item, index) => (
+    <TouchableOpacity style={styles.row} key={index} activeOpacity={0.75}>
+      <Link to={`/detail/${item}`}> 
+        <Text style={{fontSize: 20}}>Image {item}</Text>
+      </Link>
+    </TouchableOpacity>
+  ));
+
+  return <View style={styles.container}>{imgList}</View>;
+};
+
+...
+```
+
+data의 각 원소를 Link 컴포넌트로 감싼 부분만 추가되었습니다. 클릭하면 해당 DetailView로 라우팅되도록 `to` Props에 아이템 숫자를 그대로 넘겨주었습니다. 예를 들어 1번 아이템을 클릭하면 /detail/1로 라우팅 되도록 말입니다. 
+
+이제 위에서 2번 작업에 해당하는, DetailView에서 해당 id를 패러미터로 전달받을 수 있도록 작성해보겠습니다. 
+
+## DetailView 작성하기 
+
+src/components 폴더 밑에 *DetailView.js*를 생성한 다음 아래처럼 작성해주세요. 
+
+***DetailView.js***
+
+```jsx
+import React from 'react';
+import {View, StyleSheet, Text, Button} from 'react-native';
+
+const DetailView = ({match, history}) => {
+  const handleBack = () => {
+    history.goBack();
+  };
+  const id = match.params.id;
+
+  return (
+    <>
+      <View>
+        <Button onPress={handleBack} title="Back" />
+      </View>
+      <View style={styles.container}>
+        <Text style={styles.heading}>Detail {id}</Text>
+      </View>
+    </>
+  );
+};
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    alignItems: 'center',
+    flexDirection: 'column',
+  },
+  heading: {
+    marginTop: 20,
+    fontSize: 24,
+    fontWeight: '600',
+  },
+});
+
+export default DetailView;
+```
+
+라우트로 사용되는 컴포넌트라면 `match`, `history`, `location` 객체를 Props로 전달받게 됩니다. DetailView 컴포넌트는 App.js에서 확인할 수 있듯이 라우트로 설정하였고, 이에 따라 match와 history 객체에 접근할 수 있습니다. 이 Props를 받아올 때 비구조화 할당 문법을 사용했는데 이에 익숙지 않다면 Destructing Props in React 등의 키워드로 검색해보세요. 
```

Comment:
Where is the `location` object? I can't see where it is used.

Reply:
I should have written it more accurately. I've revised the description a little. Thanks!